cloudinit/config/cc_ca_certs.py

```diff
@@ -107,88 +135,108 @@ def add_ca_certs(distro_cfg, certs):
-        cr_cont = "\n".join(
-            [
-                line
-                for line in orig.splitlines()
-                if line != distro_cfg["ca_cert_filename"]
-            ]
-        )
-        out = "%s\n%s\n" % (cr_cont.rstrip(), distro_cfg["ca_cert_filename"])
+        out = ""
+        for line in orig.splitlines():
+            if line.startswith("#") or line == "":
+                out += line + "\n"
+            else:
+                out += "!" + line + "\n"
     util.write_file(distro_cfg["ca_cert_config"], out, omode="wb")
 
 
-def remove_default_ca_certs(distro_name, distro_cfg):
+def move_freebsd_ca_certs_to_blacklist_dir(distro_cfg):
     """
-    Removes all default trusted CA certificates from the system. To actually
-    apply the change you must also call L{update_ca_certs}.
+    For every file in the CA_CERT_PATH directory move it to the
+    CA_CERT_BLACKLIST_DIR in order to disable it.
 
-    @param distro_name: String providing the distro class name.
     @param distro_cfg: A hash providing _distro_ca_certs_configs function.
     """
-    util.delete_dir_contents(distro_cfg["ca_cert_path"])
-    util.delete_dir_contents(distro_cfg["ca_cert_system_path"])
-    util.write_file(distro_cfg["ca_cert_config"], "", mode=0o644)
-
-    if distro_name in ["debian", "ubuntu"]:
-        debconf_sel = (
-            "ca-certificates ca-certificates/trust_new_crts " + "select no"
-        )
-        subp.subp(("debconf-set-selections", "-"), debconf_sel)
+    if (
+        distro_cfg["ca_cert_config"] is None
+        or distro_cfg["ca_cert_blacklist_path"] is None
+    ):
+        return
+    LOG.debug("Moving system CA certificates to blacklist directory")
+    shutil.move(
+        distro_cfg["ca_cert_path"], distro_cfg["ca_cert_blacklist_path"])
 
 
 def handle(
     name: str, cfg: Config, cloud: Cloud, log: Logger, args: list
 ) -> None:
     """
-    Call to handle ca-cert sections in cloud-config file.
+    Call to handle ca_cert sections in cloud-config file.
 
-    @param name: The module name "ca-cert" from cloud.cfg
+    @param name: The module name "ca_cert" from cloud.cfg
     @param cfg: A nested dict containing the entire cloud config contents.
     @param cloud: The L{CloudInit} object in use.
     @param log: Pre-initialized Python logger object to use for logging.
     @param args: Any module arguments from cloud.cfg
     """
     if "ca-certs" in cfg:
-        log.warning(
+        LOG.warning(
             "DEPRECATION: key 'ca-certs' is now deprecated. Use 'ca_certs'"
             " instead."
         )
     elif "ca_certs" not in cfg:
-        log.debug(
+        LOG.debug(
             "Skipping module named %s, no 'ca_certs' key in configuration",
             name,
         )
         return
 
     if "ca-certs" in cfg and "ca_certs" in cfg:
-        log.warning(
+        LOG.warning(
             "Found both ca-certs (deprecated) and ca_certs config keys."
             " Ignoring ca-certs."
```

Comment:
Has nothing to do with this PR, but the warning says "ignoring ca-certs" and then below we have `ca_cert_cfg = cfg.get("ca_certs", cfg.get("ca-certs"))`.

Reply:
I'm assuming as that is a "fall through" for either (a) ca_certs section only, or (b) ca-certs section only, or (c) both ca_certs and ca-certs sections that the behaviour is that only the ca_certs settings are used (as it comes first) when both are defined.

Reply:
Not related to this PR, so resolving.